### PR TITLE
libs: Add `envoy.dependency.check[release-dates]`

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

requires #186

### CODE REVIEW

File: envoy.dependency.check/envoy/dependency/check/abstract/checker.py [hunk 1]
```diff
@@ -95,6 +145,25 @@ async def dep_cve_check(
...
             self.succeed("cves", [f"No CVEs found for: {dep.id}"])
 
+    async def dep_date_check(
+            self,
+            dep: "abstract.ADependency") -> None:
+        """Check dates for dependency."""
+        if not await dep.release.date:
+            self.error(
+                "dates",
+                [f"{dep.id} is a GitHub repository with no no inferrable "
```
Comment (kfaseela) [inline envoy.dependency.check/envoy/dependency/check/abstract/checker.py:155]: fix "no no"

Comment (phlax) [reply envoy.dependency.check/envoy/dependency/check/abstract/checker.py:155]: 8/

File: envoy.dependency.check/envoy/dependency/check/abstract/checker.py [hunk 2]
```diff
@@ -21,7 +26,15 @@ class ADependencyChecker(
         metaclass=abstracts.Abstraction):
     """Dependency checker."""
 
-    checks = ("cves", )
+    checks = ("cves", "dates")
```
Comment (kfaseela) [inline envoy.dependency.check/envoy/dependency/check/abstract/checker.py:29]: should this have a better name than simply "dates" ?

Comment (phlax) [reply envoy.dependency.check/envoy/dependency/check/abstract/checker.py:29]: yep, i think that would be better

how about `release-dates` as then its consistent with `releases` and `release-issues`

Comment (kfaseela) [reply envoy.dependency.check/envoy/dependency/check/abstract/checker.py:29]: yeah that would be good

**Review by kfaseela — APPROVED** [review]

**Pull request merged**